merlin/dag/executors.py

```diff
@@ -177,17 +178,27 @@ def _transform_data(self, node, input_data, capture_dtypes=False):
 
                 if is_list:
                     col_dtype = list_val_dtype(col_series)
-                if hasattr(col_dtype, "as_numpy_dtype"):
-                    col_dtype = col_dtype.as_numpy_dtype()
-                elif hasattr(col_series, "numpy"):
+
+                # TODO: Add a utility that condenses the known methods of fetching dtypes
+                # from series/arrays into a single function, so that Tensorflow specific
+                # code doesn't leak into the executors
```

Comment:
The need for this new utility was already latent in the existing code, but it becomes a lot clearer once you have a cross-framework way to translate dtypes. Chose not to tackle it here since that would expand the scope of this already somewhat large PR.